### PR TITLE
[Enhancement] Add `:bg` and `:man` Commands

### LINKED ISSUE #1602: [Discussion] Special Use Cases of Shell Command Execution
@fox0430

I found the following two use cases of the `:!` command being pretty common in my development workflow:
- `:!echo` to see the recent terminal output, for example from a `grep` command for lines to edit
- `:!man <something>` to view a certain manual page, especially for C standard library functions

Both can be made easily into dedicated commands on their own as they are special cases of the regular command execution.  The first one is simply `shellCommand` without line 1085 (https://github.com/fox0430/moe/blob/develop/src/moepkg/exmode.nim#L1085) and could be named `bg`, for example, as it is the opposite action to the UNIX command `fg`.  Although Moe explicitly supports `Ctrl + Z` out of the box, a dedicated internal command might avoid forgetting about a paused Moe instance in the background.

The second one is another special case of `shellCommand` without the lines 1086 and 1087 (https://github.com/fox0430/moe/blob/develop/src/moepkg/exmode.nim#L1086; https://github.com/fox0430/moe/blob/develop/src/moepkg/exmode.nim#L1087) as the editor perspective would be restored automatically when `man` returns.  The command could be named `man` and take the arguments for `man` as parameters.  As it is already officially recommended to run Moe in a Linux environment, the presence of the `man` executable could be considered a sane assumption.

I would like to discuss whether it would be useful to give those two use cases commands on their own.

### PR DESCRIPTION
**Pull request by kevinmatthes**
@fox0430

This Pull Requests fixes #1602.

- The new commands are implemented.
- The new commands are documented in the help.
- The new commands are documented in the command suggestion.
- The new commands are documented in the "How to Use" command overview.

I just need to update my Scriv fragment, then, the PR is ready for merging.

### CODE REVIEW

File: src/moepkg/exmode.nim [hunk 1]
```diff
@@ -237,6 +237,13 @@ proc isForceQuitCommand(command: seq[seq[Rune]]): bool {.inline.} =
 proc isShellCommand(command: seq[seq[Rune]]): bool {.inline.} =
   return command.len >= 1 and command[0][0] == ru'!'
 
+proc isBackgroundCommand(command: seq[seq[Rune]]): bool {.inline.} =
+  return command.len == 1 and cmpIgnoreCase($command[0], "bg") == 0
+
+proc isManualCommand(command: seq[seq[Rune]]): bool {.inline.} =
+  # TODO:  Configure a default manual page to show on `:man`.
+  return command.len > 1 and cmpIgnoreCase($command[0], "man") == 0
```
Comment (kevinmatthes) [inline src/moepkg/exmode.nim:245]: @fox0430

`command.len >= 1` allows for `:man` and thus the configuration of a default manual page to show.  I would like to suggest to either redirect to `man`'s manual page or Moe's one (as soon as it was written).

This is opted out for now but can be changed very easily.

File: src/moepkg/exmode.nim [hunk 2]
```diff
@@ -1091,6 +1098,26 @@ proc shellCommand(status: var EditorStatus, shellCommand: string) =
...
+  restoreTerminalModes()
+  status.commandLine.clear
+  status.changeMode(currentBufStatus.prevMode)
+
+proc manualCommand(status: var EditorStatus, manualInvocationCommand: string) =
+  saveCurrentTerminalModes()
+  exitUi()
+
+  # TODO:  Configure a default manual page to show on `:man`.
+  discard execShellCmd(manualInvocationCommand)
```
Comment (kevinmatthes) [inline src/moepkg/exmode.nim]: @fox0430

Here, the default manual page can be configured.  If the `manualInvocationCommand` equals `man`, the default page could be shown.  Else, there are arguments to be passed to `man`.

File: src/moepkg/commandlineutils.nim [hunk 3]
```diff
@@ -45,6 +46,7 @@ const exCommandList: array[65, tuple[command, description: string]] = [
   (command: "log", description: "                  | Open the log viewer"),
   (command: "ls", description: "                   | Show the all buffer"),
   (command: "lsw", description: "                  | Show the all workspace"),
+  (command: "man", description: "                  | Show the given UNIX manual page, if available"),
```
Comment (kevinmatthes) [inline src/moepkg/commandlineutils.nim:49]: @fox0430

`man` will write an error message to the terminal if the queried page does not exist.  The user shall be responsible for calling only existing manual pages.

File: src/moepkg/exmode.nim (hunk 2 above)
Comment (fox0430) [inline src/moepkg/exmode.nim]: If it fails, I think it's better to display an error message.

```suggestion
  let exitCode = execShellCmd(manualInvocationCommand)
  restoreTerminalModes()

  if exitCode != 0:
    let mess = "Error: No manual entry for " & manualInvocationCommand
    status.commandLine.writeMessageOnCommandLine(mess, EditorColorPair.errorMessage)
    status.messageLog.add(mess.toRunes)
  else:
    status.commandLine.clear
```

**Review by fox0430 — CHANGES_REQUESTED** [review]

File: src/moepkg/exmode.nim (hunk 2 above)
Comment (kevinmatthes) [reply src/moepkg/exmode.nim]: @fox0430

I applied the requested changes such that the error message will now be shown on the editor's command line, as well. :wrench: :heavy_check_mark:

**Review by fox0430 — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**